packages/widget-v2/src/pages/SwapExecutionPage/useFetchTransactionStatus.tsx

```diff
@@ -24,86 +36,97 @@ export const useFetchTransactionStatus = () => {
+      return "signing";
     }
+    if (!overallState) return "";
 
-    if (
-      operationTransferEventsArray.find(({ status }) => status === "failed")
-    ) {
-      return "failed";
-    }
-    if (
-      operationTransferEventsArray.find(({ status }) => status === "pending")
-    ) {
-      return "pending";
-    }
-    if (
-      operationTransferEventsArray.every(
-        (state) => state.status === "broadcasted"
-      )
-    ) {
-      return "broadcasted";
-    }
-  }, [isPending, operationToTransferEventsMap, route?.operations, setOverallStatus]);
-
-  useEffect(() => {
-    if (overallStatus === "completed" || overallStatus === "failed") return;
+    return getSimpleOverallStatus(overallState);
+  }, [isPending, operationTransferEventsArray.length, transactionStatus]);
 
+  const updateOperationToTransferEventsMap = useCallback(() => {
     const transferEvents =
       getTransferEventsFromTxStatusResponse(transactionStatus);
-    const operationToTransferEventsMap = getOperationToTransferEventsMap(
-      transactionStatus ?? [],
-      clientOperations
-    );
-    const operationTransferEventsArray = Object.values(
-      operationToTransferEventsMap
-    );
 
     if (transactionDetailsArray.length === 1 && transferEvents.length === 0) {
+      // temporarily setting operationToTransferEventsMap to have the first
+      // operation pending before the API returns any transfer events
+      // so that the UX is nicer
       setOperationToTransferEventsMap({
         0: {
           status: "pending",
         } as ClientTransferEvent,
       });
-    }
-    if (operationTransferEventsArray.length > 0) {
+    } else if (operationTransferEventsArray.length > 0) {
+      // setting operationToTransferEventsMap if operationToTransferEventsMap
+      // is retrieved properly from getOperationToTransferEventsMap
       setOperationToTransferEventsMap(operationToTransferEventsMap);
+    } else if (simpleOverallStatus) {
+      // manually creating and setting operationToTransferEventsMap for
+      // the case that we have an overallStatus completed or failed
+      // but no transfer events (ie. swaps that remain on chain)
+      if (["completed", "failed"].includes(simpleOverallStatus)) {
+        const derivedOperationToTransferEventsMap = clientOperations.reduce(
+          (accumulator, operation, index) => {
+            accumulator[index] = {
+              status: simpleOverallStatus,
+              state:
+                simpleOverallStatus === "completed"
+                  ? "TRANSFER_SUCCESS"
+                  : "TRANSFER_FAILURE",
+              toChainID: operation.toChainID,
+              fromChainID: operation.fromChainID,
+            } as ClientTransferEvent;
+
+            return accumulator;
+          },
+          {} as Record<number, ClientTransferEvent>
+        );
+        setOperationToTransferEventsMap(derivedOperationToTransferEventsMap);
+      }
```

Comment:
handling the case of not getting transfer event back but getting an overall status

Reply:
it won't render an explorerLink for now but @NotJeremyLiu says this is fine for the time being and that V1 also doesn't render links for this case?

Reply:
also state toChainID and fromChainID honestly aren't really used at all typescript just complaining if i dont provide it